lib/Runtime/Library/JavascriptObject.cpp

```diff
@@ -2184,3 +2171,27 @@ void JavascriptObject::SpreadObjectLiteral(Var source, Var to, ScriptContext* sc
     AssertOrFailFast(succeeded);
     CopyDataPropertiesHelper<false, false>(source, target, scriptContext);
 }
+
+void JavascriptObject::Restify(Var source, Var to, Var excludedStatic, Var excludedComputed, ScriptContext* scriptContext)
+{
+    RecyclableObject* target = nullptr;
+    bool succeeded = JavascriptConversion::ToObject(to, scriptContext, &target);
+    AssertOrFailFast(succeeded);
+
+    const Js::PropertyIdArray* propIdsStatic = (const Js::PropertyIdArray*)(excludedStatic);
+    const Js::PropertyIdArray* propIdsComputed = (const Js::PropertyIdArray*)(excludedComputed);
```

Comment:
Seems like most of the time excludedComputed will be empty. Do you think it might be useful to avoid allocating it and pass null in that case? If doing so is more trouble than it's worth or introduces dangerous complexity, it's fine as-is.

Reply:
By the way, I like how you split the static and computed property IDs so the static ones can go in the auxiliary data and not need copying around. That's great.

---
In reply to: [204246798](https://github.com/Microsoft/ChakraCore/pull/5498#discussion_r204246798) [](ancestors = 204246798)

Reply:
Good point. I actually was going to implement this later as I noticed this while addressing other comments haha.

---
In reply to: [204248858](https://github.com/Microsoft/ChakraCore/pull/5498#discussion_r204248858) [](ancestors = 204248858,204246798)